### DISCUSSION TITLE
一分钟读论文：《工程师如何设计测试用例：深入观察性研究》

### DISCUSSION THREAD

**Discussion by giscus** [unbug/unbug.github.io#5] [Announcements]
# 一分钟读论文：《工程师如何设计测试用例：深入观察性研究》

先写测试还是先写代码？荷兰代尔夫特特理工大学和澳大利亚墨尔本大学的论文《How Developers Engineer Test Cases: An Observational Study》 观察了 13 位具有不同级别经验的工程师为真实世界的开源码编写测试用例，并调查了 72 位工程师之后，总结了一个框架和一组策略来解释工程师如何设计测试用例。

https://unbug.github.io/How-Developers-Engineer-Test-Cases-An-Observational-Study/

**Comment by unbug** [comment]
大部分时间浪费在“测试失败后立即调试” 有点意思
